src/Compiler/NodeVisitor/TransformForeachWithIteratorNodeVisitor.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Efabrica\PHPStanLatte\Compiler\NodeVisitor;
+
+use Efabrica\PHPStanLatte\Compiler\NodeVisitor\Behavior\ActualClassNodeVisitorBehavior;
+use PhpParser\Node;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\New_;
+use PhpParser\Node\Stmt\Expression;
+use PhpParser\Node\Stmt\Foreach_;
+use PhpParser\NodeVisitorAbstract;
+
+/**
+ * changed output from:
+ * <code>
+ * foreach ($iterator = $ʟ_it = new LR\CachingIterator($stringList, $ʟ_it ?? null) as $string)
+ * </code>
+ *
+ * to:
+ * <code>
+ * $iterator = $ʟ_it = new LR\CachingIterator($stringList, $ʟ_it ?? null)
+ * foreach ($stringList as $string) {
+ * </code>
+ */
+final class TransformForeachWithIteratorNodeVisitor extends NodeVisitorAbstract implements PostCompileNodeVisitorInterface
+{
+    use ActualClassNodeVisitorBehavior;
+
+    /**
+     * @return Node[]|null
+     */
+    public function leaveNode(Node $node): ?array
+    {
+        if (!$node instanceof Foreach_) {
+            return null;
+        }
+
+        if (!$node->expr instanceof Assign) {
+            return null;
+        }
+
+        if (!$node->expr->expr instanceof Assign) {
```

Comment:
Maybe use while loop here to find first node that is not `Assign` expression? I am not sure there are/will be always exactly two of them.

Reply:
That depends on latte compiler. Now there are two of them always

Reply:
I would preffer more robust solution but I think it can wait if issue with different count occur